tests/README.md

```diff
@@ -0,0 +1,427 @@
+# Automated regression tests for librdkafka
+
+
+## Supported test environments
+
+While the standard test suite works well on OSX and Windows,
+the full test suite (which must be run for PRs and releases) will
+only run on recent Linux distros due to its use of ASAN, Kerberos, etc.
+
+
+## Automated broker cluster setup using trivup
+
+A local broker cluster can be set up using
+[trivup](https://github.com/edenhill/trivup), which is a Python package
+available on PyPi.
+These self-contained clusters are used to run the librdkafka test suite
+on a number of different broker versions or with specific broker configs.
+
+trivup will download the specified Kafka version into its root directory,
+the root directory is also used for cluster instances, where Kafka will
+write messages, lgos, etc.
+The trivup root directory is by default `tmp` in the current directory but
+may be specified by setting the `TRIVUP_ROOT` environment variable
+to alternate directory, e.g., `TRIVUP_ROOT=$HOME/trivup make full`.
+
+First install trivup:
+
+    $ pip install trivup
+
+Bring up a Kafka cluster (with the specified version) and start an interactive
+shell, when the shell is exited the cluster is brought down and deleted.
+
+    $ ./interactive_broker_version.py 2.3.0   # Broker version
+
+In the trivup shell, run the test suite:
+
+    $ make
+
+
+If you'd rather use an existing cluster, you may omit trivup and
+provide a `test.conf` file that specifies the brokers and possibly other
+librdkafka configuration properties:
+
+    $ cp test.conf.example test.conf
+    $ $EDITOR test.conf
+
+
+
+## Run specific tests
+
+To run tests:
+
+    # Run tests in parallel (quicker, but harder to troubleshoot)
+    $ make
+
+    # Run a condensed test suite (quickest)
+    # This is what is run on CI builds.
+    $ make quick
+
+    # Run tests in sequence
+    $ make run_seq
+
+    # Run specific test
+    $ TESTS=0004 make
+
+    # Run test(s) with helgrind, valgrind, gdb
+    $ TESTS=0009 ./run-test.sh ./merged valgrind|helgrind|gdb
```

Comment:
but how do you integrate that with VS Code for a beautiful interactive debugging experience?